addons/mail/static/src/components/attachment_delete_confirm_dialog/attachment_delete_confirm_dialog.js

```diff
@@ -0,0 +1,88 @@
+odoo.define('mail/static/src/components/attachment_delete_confirm_dialog/attachment_delete_confirm_dialog.js', function (require) {
+'use strict';
+
+const useStore = require('mail/static/src/component_hooks/use_store/use_store.js');
+
+const Dialog = require('web.OwlDialog');
+
+const { Component } = owl;
+const { useRef } = owl.hooks;
+
+class AttachmentDeleteConfirmDialog extends Component {
+
+    /**
+     * @override
+     */
+    constructor(...args) {
+        super(...args);
+        useStore(props => {
+            const attachment = this.env.models['mail.attachment'].get(props.attachmentLocalId);
+            return {
+                attachment: attachment ? attachment.__state : undefined,
+            };
+        });
+        // to manually trigger the dialog close event
+        this._dialogRef = useRef('dialog');
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * @returns {mail.attachment}
+     */
+    get attachment() {
+        return this.env.models['mail.attachment'].get(this.props.attachmentLocalId);
+    }
+
+    /**
+     * @returns {string}
+     */
+    get CONFIRMATION() {
+        return this.env._t("Confirmation");
+    }
+
+    /**
+     * @returns {string}
+     */
+    getText() {
+        return _.str.sprintf(
+            this.env._t("Do you really want to delete '%s'?"),
+            owl.utils.escape(this.attachment.displayName)
+        );
+    }
```

Comment:
Would be easier to read JS/template by naming functions with "title"/"body".

Reply:
I used the same pattern as `moderation_discard_dialog.js`, so should we change all those using this patterns ?

Reply:
No, just this one. I had several remarks on this PR, so I took opportunity to also point this one.

Reply:
Hm... if it's fast and not problematic, could do it anyway.